pkg/pillar/cmd/downloader/syncop.go

```diff
@@ -175,6 +176,7 @@ func handleSyncOp(ctx *downloaderContext, key string,
 	}
 
 	// Loop through all interfaces until a success
+	var gotNoSuitableAddress bool
```

Comment:
Is there a behavioral difference between this code using a boolean and the send.go code using a counter for the same purpose? Can we make them use the same approach reduce the number of variants the reader needs to understand?

Reply:
i think the difference is for download, its not just a error spit out in the log, it forms an error-String to be used to publish for this download failure reason. I want to mainly reduce the size of that string, thus for multiple interfaces, if we got the same error, we just form one error-string to be used for publish, but the logging is not suppressed on this, similar to the send.go